pulpcore/cli/core/group.py

```diff
@@ -217,11 +231,19 @@ def role(ctx: click.Context, pulp_ctx: PulpCLIContext, group_ctx: PulpGroupConte
             click.option(
                 "--object",
                 "content_object",
-                required=True,
+                default="",
                 callback=lookup_callback("content_object", PulpGroupRoleContext),
                 expose_value=False,
                 help=_('Associated object; use "" for global assignments.'),
             ),
+            pulp_option(
+                "--domain",
+                # This only accepts HREFs
+                callback=lookup_callback("domain", PulpGroupRoleContext),
```

Comment:
```suggestion
                callback=lookup_callback("domain", PulpDomainContext),
```
does this help to revisit you comment about HREFs?

Reply:
No, the `lookup_callback` only adds the field to the entity context's lookup and requires that the context exists in the command, which `PulpDomainContext` does not.

 I added more functionality to `resource_option` to allow for specifying the domain name for this command.